Tests/Server/JokesServiceTests.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+using FluentAssertions;
+using Moq;
+using JokesOnYou.Web.Api.Repositories.Interfaces;
+using JokesOnYou.Web.Api.Services;
+using JokesOnYou.Web.Api.Models;
+using JokesOnYou.Web.Api.DTOs;
+using AutoMapper;
+using JokesOnYou.Web.Api.Profiles;
+
+namespace Tests.Server
+{
+    public class JokesServiceTests
+    {
+        private readonly IMapper _mapper = new Mapper(new MapperConfiguration(cfg => cfg.AddProfile<JokesProfile>()));
+
+        [Fact]
+        public void GetAllUsers_Returns_ListOfJokesDto()
+        {
+            var repo = new Mock<IJokesRepository>();
+            var service = new JokesService(repo.Object, _mapper);
+            List<Joke> Jokes = new List<Joke>() {
+                new Joke() { Id = 1, Premise = "1", Punchline = "1", Likes = 1, Dislikes = 0},
+                new Joke() { Id = 2, Premise = "2", Punchline = "2", Likes = 2, Dislikes = 0}
+            };
+            repo.Setup(r => r.GetAllJokesAsync().Result).Returns(Jokes);
+            List<JokeReplyDto> expected = new List<JokeReplyDto>();
+            foreach (Joke joke in Jokes)
+            {
+                expected.Add(new JokeReplyDto() { Id = joke.Id, Premise = joke.Premise, Punchline = joke.Punchline, Likes = joke.Likes, Dislikes = joke.Dislikes });
+            }
+
+            var actualList = service.GetAllJokesAsync().Result;
```

Comment:
How about awaiting this async method instead of trying to get the result out?

Reply:
https://stackoverflow.com/questions/29809302/is-there-a-difference-between-calling-result-or-await-on-known-completed-tasks#:~:text=The%20only%20functional%20difference%20that,will%20just%20raise%20the%20exception.

So i think not much difference but await needs the whole test to be async. but ill test it out anyway would like to see.

Reply:
> https://stackoverflow.com/questions/29809302/is-there-a-difference-between-calling-result-or-await-on-known-completed-tasks#:~:text=The%20only%20functional%20difference%20that,will%20just%20raise%20the%20exception.
> 
> So i think not much difference but await needs the whole test to be async. but ill test it out anyway would like to see.

actually theres more to it, thats why async/await exists. https://stackoverflow.com/questions/58312479/is-accessing-async-result-considered-as-bad-practice

I think theres a ReturnsAsync() method aswell for Moq so you should be able to figure it out np

Reply:
So the Service doesn't do anything anymore. what should be tested is the mapper now ;)
removed the test.